[FEATURE-REQUEST] Allow to pass no arguments to the constructor

**Is your feature request related to a problem? Please describe.**
The following code
```typescript
import { RPC } from 'rpc-request';
const rpc = new RPC();
```
produces an error:
```
error TS2554: Expected 1 arguments, but got 0.
```


**Describe the solution you'd like**
Allow to pass no arguments to the constructor

**Describe alternatives you've considered**
```typescript
import { RPC } from 'rpc-request';
const rpc = new RPC({});
```

--- a/index.ts
+++ b/index.ts
@@ -1,68 +1,70 @@
 import * as request from 'request-promise-native';
 
+export type RPCOtpions = request.RequestPromiseOptions | request.Options;
+
 export class RPC {
-  _rpoptions: request.RequestPromiseOptions;
+  _rpoptions: RPCOtpions;
   /**
    * @params {Object} [options]
    */
-  constructor({ ...options }: request.RequestPromiseOptions) {
+  constructor(options: RPCOtpions = {}) {
     this._rpoptions = options;
   }
 
   /**
-   * @params {Object} options
+   * @params {Object} [options]
    */
-  async get({ ...options }: request.Options) {
+  async get(options: RPCOtpions = {}) {
     return this.request({ ...options, method: 'GET' });
   }
 
   /**
-   * @params {Object} options
+   * @params {Object} [options]
    */
-  async post({ ...options }: request.Options) {
+  async post(options: RPCOtpions = {}) {
     return this.request({ ...options, method: 'POST' });
   }
 
   /**
-   * @params {Object} options
+   * @params {Object} [options]
    */
-  async put({ ...options }: request.Options) {
+  async put(options: RPCOtpions = {}) {
     return this.request({ ...options, method: 'PUT' });
   }
 
   /**
-   * @params {Object} options
+   * @params {Object} [options]
    */
-  async patch({ ...options }: request.Options) {
+  async patch(options: RPCOtpions = {}) {
     return this.request({ ...options, method: 'PATCH' });
   }
 
   /**
-   * @params {Object} options
+   * @params {Object} [options]
    */
-  async delete({ ...options }: request.Options) {
+  async delete(options: RPCOtpions = {}) {
     return this.request({ ...options, method: 'DELETE' });
   }
 
   /**
-   * @params {Object} options
+   * @params {Object} [options]
    */
-  async head({ ...options }: request.Options) {
+  async head(options: RPCOtpions = {}) {
     return this.request({ ...options, method: 'HEAD' });
   }
 
   /**
-   * @params {Object} options
+   * @params {Object} [options]
    */
-  async options({ ...options }: request.Options) {
+  async options(options: RPCOtpions = {}) {
     return this.request({ ...options, method: 'OPTIONS' });
   }
 
   /**
-   * @params {Object} options
+   * @params {Object} [options]
    */
-  async request({ ...options }: request.Options) {
-    return this.defaults(options);
+  async request(options: RPCOtpions = {}) {
+    return this.defaults(RPC.prepareOptions(options, this._rpoptions));
   }
 
   /**
@@ -83,12 +85,32 @@ export class RPC {
   }
 
   /**
-   * @params {Object} options
+   * @params {Object} [options]
    */
   static defaults(
-    options: request.RequestPromiseOptions
+    options: request.RequestPromiseOptions = {}
   ): request.RequestPromiseAPI {
     return request.defaults(options);
+  }
+
+  /**
+   * @params {Object} options
+   * @params {Object} rpoptions
+   */
+  static prepareOptions(
+    options: RPCOtpions,
+    rpoptions: RPCOtpions
+  ): request.Options {
+    if (!('url' in options || 'uri' in options)) {
+      if (!('url' in rpoptions || 'uri' in rpoptions)) {
+        throw new Error('options.uri is a required argument');
+      }
+      if ('uri' in rpoptions) {
+        return { ...options, uri: rpoptions.uri };
+      }
+      return { ...options, uri: rpoptions.url };
+    }
+    return options;
   }
 
   get defaults(): request.RequestPromiseAPI {

